patches/api/0469-Added-API-to-get-player-s-proxy-address.patch

```diff
@@ -0,0 +1,27 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: nostalfinals <yuu8583@proton.me>
+Date: Mon, 8 Apr 2024 23:24:38 +0800
+Subject: [PATCH] Added API to get player's proxy address
+
+
+diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
+index c6cb4f17469a8f2e60dd3e28d41402851ce5fb21..85887b7ec9f7b6313a1e5752ae0326a9d2a5f9ea 100644
+--- a/src/main/java/org/bukkit/entity/Player.java
++++ b/src/main/java/org/bukkit/entity/Player.java
+@@ -248,6 +248,16 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
+     @Nullable
+     public InetSocketAddress getAddress();
+ 
++    // Paper start - Add API to get player's proxy address
++    /**
++    * Gets the socket address of this player's proxy
++    *
++    * @return the player's proxy address, null if the server doesn't have Proxy Protocol enabled, or the player didn't connect to an HAProxy instance
++    */
++    @Nullable
++    public InetSocketAddress getProxyAddress();
```

Comment:
This should generally be more declarative to what this is actually exposing, as it generally feels detached from what it actually exposes

Reply:
getProxyJoinedWithAddress() or is that even worse?

Reply:
I'm considering if just name it as `getHAProxyAddress` directly will be better.

Reply:
I like it too 👍